plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -131,56 +146,88 @@ public boolean equals(Object obj)
             return false;
         }
         DeltaLakeColumnHandle other = (DeltaLakeColumnHandle) obj;
-        return Objects.equals(this.name, other.name) &&
-                Objects.equals(this.type, other.type) &&
-                Objects.equals(this.fieldId, other.fieldId) &&
-                Objects.equals(this.physicalName, other.physicalName) &&
-                Objects.equals(this.physicalType, other.physicalType) &&
-                this.columnType == other.columnType;
+        return Objects.equals(this.baseColumnName, other.baseColumnName) &&
+                Objects.equals(this.baseType, other.baseType) &&
+                Objects.equals(this.baseFieldId, other.baseFieldId) &&
+                Objects.equals(this.basePhysicalColumnName, other.basePhysicalColumnName) &&
+                Objects.equals(this.basePhysicalType, other.basePhysicalType) &&
+                this.columnType == other.columnType &&
+                Objects.equals(this.projectionInfo, other.projectionInfo);
+    }
+
+    @JsonIgnore
+    public String getColumnName()
+    {
+        checkState(isBaseColumn(), "Unexpected dereference: %s", this);
+        return baseColumnName;
+    }
+
+    @JsonIgnore
+    public String getQualifiedName()
+    {
+        return projectionInfo.map(projectionInfo -> basePhysicalColumnName + "#" + projectionInfo.getPartialName())
```

Comment:
`getQualifiedName` returns the physical name always.

Reply:
CC: @findepi | @findinpath

Reply:
`getQualified*Physical*Name` ?